pkg/cmd/grafana-cli/utils/command_line.go

```diff
@@ -53,6 +58,20 @@ func (c *ContextCommandLine) PluginDirectory() string {
 }
 
 func (c *ContextCommandLine) PluginRepoURL() string {
+	if c.ConfigFile() != "" && !slices.Contains(c.FlagNames(), "repo") {
```

Comment:
Would like to see a comment here describing the behaviour.

Also it might be nicer to have "default" behaviour first. For example:
```suggestion
    repoURL := c.String("repo")
    if repoURL != "" {
      return repoURL
    }
	if c.ConfigFile() != "" {
```

Reply:
for example like this https://github.com/grafana/grafana/pull/86578/commits/3ace37b93008a83a54a2efbe991f6e26075a3f69

Reply:
Yep that makes sense - I would probably note that important detail in the comment (that the slice check is not the same as just checking repo). Otherwise (but probably good idea anyway) to add a test for that 😉

Reply:
alright, down the testing rabbit hole - would maybe need some help here with all those mocks

https://github.com/grafana/grafana/pull/86578/commits/cce302261d50e3a10e4652810251fb7781993d4c is this good enough?

Reply:
Awesome stuff - that's good enough for me 🚀